### PR TITLE
Documentation and sample code changes for torch_ort_inference

### PR DESCRIPTION
**Pull request by saipj**
- Readme and Usage document changes
- Sample code changes to fix input data validations
- Using cached variable for graph input_info

### CODE REVIEW

**Review by natke — APPROVED** [review]

File: torch_ort_inference/docs/usage.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```python
+provider_options = OpenVINOProviderOptions(backend = "GPU", precision = "FP16")
+model = ORTInferenceModule(model, provider_options = provider_options)
+```
+Supported backend-precision combinations:
+| Backend | Precision |
+| --------| --------- |  
+|   CPU   |    FP32   |
+|   GPU   |    FP32   |
```
Comment (askhade) [inline torch_ort_inference/docs/usage.md:24]: Are all these supported and tested? I thought this release was only targeting CPU

Comment (saipj) [reply torch_ort_inference/docs/usage.md:24]: Yes. All the above configurations are supported and tested in our pipelines.

File: torch_ort_inference/docs/usage.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+## Essential APIs
+
+To add the OpenVINO™ integration with TorchORT package to your PyTorch application, add following 2 lines of code:
+
+```python
+from torch_ort import ORTInferenceModule
+model = ORTInferenceModule(model)
+```
+
+By default, CPU backend with FP32 precision is enabled. You can set different backend and supported precision using OpenvINOProviderOptions as below:
```
[the remark this answers is not in the document]
Comment (saipj) [reply torch_ort_inference/docs/usage.md]: Updated.

File: torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py [hunk 3]
```diff
@@ -106,8 +106,7 @@ def _forward_call(self, *inputs, **kwargs):
 
         # Use IO binding
         onnx_input_names = [inp.name for inp in self._onnx_models.exported_model.graph.input]
-        input_info = _io.parse_inputs_for_onnx_export(self._module_parameters, None, schema, inputs, kwargs)
-        inputs = _utils_infer.get_user_inputs(onnx_input_names, input_info, inputs, kwargs, self._device)
+        inputs = _utils_infer.get_user_inputs(onnx_input_names, self._flattened_module._input_info, inputs, kwargs, self._device)
```
Comment (baijumeswani) [inline torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py:109]: Where are we setting `self._flattened_module._input_info`?

Comment (baijumeswani) [reply torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py:109]: ok, I see it is being set in `_export_model`. Nevermind. Thanks for making this change.

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**